pyfstat/make_sfts.py

```diff
@@ -134,6 +122,55 @@ def basic_setup(self):
         )
         self.IFOs = ",".join(['"{}"'.format(d) for d in IFOs])
 
+    def _get_setup_from_noiseSFTs(self):
+        noise_multi_sft_catalog = lalpulsar.GetMultiSFTCatalogView(
+            lalpulsar.SFTdataFind(self.noiseSFTs, self.SFTconstraint)
+        )
+
+        # Extract information from the SFTs themselves
+        IFOs = []
+        tstart = []
+        tend = []
+        # SWIG-LAL does not import this functionality
+        gps_to_float = lambda x: x.gpsSeconds + x.gpsNanoSeconds
+
+        for catalog in noise_multi_sft_catalog.data:
+            ifo_name = lalpulsar.ListIFOsInCatalog(catalog).data[0]
+            time_stamps = lalpulsar.TimestampsFromSFTCatalog(catalog)
+
+            IFOs.append(ifo_name)
+            tstart.append(gps_to_float(time_stamps.data[0]))
+            tend.append(gps_to_float(time_stamps.data[-1]))
+
+        self.tstart = min(tstart)
+        self.tend = max(tend)
+        self.duration = self.tend - self.tstart
+        self.IFOs = ",".join(['"{}"'.format(d) for d in IFOs])
+
+        # Split noiseSFTs to account for multiple sets at once
+        noise_sft_sets = self.noiseSFTs.split(";")
+        self.sftfilepath = ""
+        for sft_set in noise_sft_sets:
+            self.sftfilepath += ";".join(glob.glob(sft_set)) + ";"
+        self.sftfilepath = self.sftfilepath[:-1]
+
+        self.sftfilenames = [
```

Comment:
Ok, my mistake: This does NOT refer to the noise SFTs, but to the output produced by MFD, which uses the noiseSFTs.

Makes total sense!

Reply:
SFT name is constructed in the same way as [MFD_V5] (loop over ifos, longest time span per each ifo).

I'll have to change some tests to make sure it works